### PR TITLE
fix: numpy version

### PR DESCRIPTION
**Pull request by SteBaum**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

Fixes numpy version compatibility with matplotlib when installing with pip.

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ psycopg2-binary = { version = "^2.9.9", optional = true }
 pymysql = { version = "^1.1.0", optional = true }
 alembic = "^1.13.1"
 alembic-postgresql-enum = "^1.2.0"
+numpy = "<2.0.0"
```
Comment (rpignolet) [inline pyproject.toml:34]: This is an optional dependency and enable only with `visualization` extras. `numpy` is used by `matplotlib` but when installed, `networkx` will use it and need a version `<2.0.0`.

File: poetry.lock [hunk 2]
```diff
@@ -968,47 +968,47 @@ scipy = ["scipy"]
 
 [[package]]
 name = "numpy"
-version = "1.26.3"
+version = "1.26.4"
```
Comment (rpignolet) [inline poetry.lock:971]: Why `numpy` is upgraded ? Did you run `poetry lock --no-update` ?

Comment (SteBaum) [reply poetry.lock:971]: done

Comment (rpignolet) [reply poetry.lock:971]: I do not see any change.

Comment (SteBaum) [reply poetry.lock]: yes I did a `poetry lock --no-update` but when I add numpy it with a poetry command it upgrades it. I added it manually and now it should be good.

**Review by rpignolet — APPROVED** [review]

**Comment by PaulFarault** [comment]
For reference, this was not a compatibility issue between `matplotlib` and `numpy` but between `networkx` and `numpy`.

`networkx` `2.5.1` is not compatible with `numpy` `2.0`. The following is enough to make it break:

```sh
pip install 'networkx[numpy]==2.5.1'
```

`numpy` is an optional dependency for `networkx` but required by `matplotlib`. Hence, installing `matplotlib` through `pip install .[visualization]` will break `networkx`.

The issue has been solved from `networkx` 3.3: https://github.com/networkx/networkx/pull/7390

**Pull request merged**